src/scripts/genXplatLttng.py

```diff
@@ -239,158 +267,193 @@ def generateLttngHeader(providerName,allTemplates,eventNodes):
+    
+    for paramName in fnSig.paramlist:
+        fnparam     = fnSig.getParam(paramName)
+        paramname   = fnparam.name
+
+        if fnparam.winType == "win:UnicodeString":
+            result.append("    INT " + paramname + "_path_size = -1;\n")
+            result.append("    PathCharString " + paramname + "_PS;\n")
+            result.append("    INT " + paramname + "_full_name_path_size")
+            result.append(" = (PAL_wcslen(" + paramname + ") + 1)*sizeof(WCHAR);\n")
+            result.append("    CHAR* " + paramname + "_full_name = ")
+            result.append(paramname + "_PS.OpenStringBuffer(" + paramname + "_full_name_path_size );\n")
+            result.append("    if (" + paramname + "_full_name == NULL )")
+            result.append("    { return ERROR_WRITE_FAULT; }\n")
+
+    result.append("\n")
+
+    #emit tracepoints
+    fnSig   = template.signature
+        
+    if not shouldPackTemplate(template):
+        linefnbody = ["    do_tracepoint(%s,\n        %s" % (providerName, eventName)]
+
+        for params in fnSig.paramlist:
+            fnparam     = fnSig.getParam(params)
+            wintypeName = fnparam.winType
+            winCount    = fnparam.count
+            paramname   = fnparam.name
+            ctf_type    = ctfDataTypeMapping.get(winCount)
+
+            line = "        "
+            if not ctf_type:
+                ctf_type    = ctfDataTypeMapping[wintypeName]
+
+            if ctf_type == "ctf_string" and wintypeName == "win:UnicodeString":
+                #emit code to convert unicode to ansi string
+
+                result.append("    " + paramname+ "_path_size = WideCharToMultiByte( CP_ACP, 0, ")
+                result.append(paramname + ", -1, ")
+                result.append(paramname + "_full_name, ")
+                result.append(paramname + "_full_name_path_size, NULL, NULL );\n")
+
+                result.append("    _ASSERTE(" +paramname+ "_path_size < " )
+                result.append(paramname + "_full_name_path_size );\n    ")
+
+                result.append(paramname + "_PS.CloseBuffer(" + paramname + "_path_size );\n")
+                result.append("    if( " + paramname + "_path_size == 0 ){ return ERROR_INVALID_PARAMETER; }\n")
+
+                line += paramname + "_full_name"
+                linefnbody.append(line)
+                continue
+
+            elif ctf_type == "ctf_sequence" or wintypeName == "win:Pointer":
+                line += "(" + lttngDataTypeMapping[wintypeName]
+                if not lttngDataTypeMapping[winCount] == " ":
+                    line += lttngDataTypeMapping[winCount]
+
+                line += ") "
+                linefnbody.append(line + paramname)
+
+            else:
+                linefnbody.append(line + paramname)
+
+        linefnbody = ",\n".join(linefnbody) + ");\n"
+        result.append(linefnbody)
+        return ''.join(result)
+
+    else:
+        header = """
+    char stackBuffer[%s];
```

Comment:
PathCharString is for strings.  It does things like null terminate the string and such.  It _could_ be adapted for this purpose, but it's not the right tool here.  We are building a raw buffer which includes strings and other data.  There didn't seem to be a good already existing construct to use, so I simply resized the array on demand.

Keep in mind though that 99% of the time my buffer code will never be touched.  The new code generation python script makes a best estimate as to how big the buffer needs to be (and is accurate for all fixed sized constructs).  The only time we'd need to resize a buffer is if the struct contains a string and that string is bigger than how much I estimate the average string size is...so it's largely academic.